### PR TITLE
[Feat][Qwen3-TTS] Make talker MTP sampling deterministic under CUDA Graphs

### PR DESCRIPTION
**Pull request by sphinxkkkbc**
PLEASE FILL IN THE PR DESCRIPTION HERE.

## Purpose
Currently, `talker_mtp` is captured without a request-specific generator or sampling-noise input. Under Full CUDA Graph execution, the request seed is therefore not applied to residual code sampling, which makes the output non-reproducible.

This PR introduces a runner-owned `sampling_noise` tensor for CUDA Graph execution. Before each graph replay, the runner fills this tensor with exponential noise using the request-local generator initialized from the request seed. `talker_mtp` then samples with vLLM's `sample_with_exponential_noise` operation.

Eager execution continues to use the generator-based sampling path. The added randomness is confined to `talker_mtp` and does not affect `code2wav`.

## Test Plan
Test and Benchmarked on RTX3090+CUDA13.0

- Verify that the test runner generates deterministic `sampling_noise` when seeds are provided.
- E2E determinism check for `Qwen3-TTS` using the same seed.
- A/B benchmark comparing this PR against `main`.

### Test command:
```
python3 -m pytest vllm-omni/tests/worker/test_omni_gpu_model_runner.py
python3 -m pytest vllm-omni/tests/e2e/online_serving/test_qwen3_tts_base.py
```
### Benchmark Command:
```
python3 vllm-omni/benchmarks/tts/bench_tts.py \
  --host 127.0.0.1 \
  --port 8091 \
  --model Qwen/Qwen3-TTS-12Hz-1.7B-Base \
  --locale en \
  --dataset-path /path/to/seedtts_testset \
  --num-prompts 128 \
  --concurrency 1 4 8 16 32 64 \
  --output-dir vllm-omni/results/qwen3-tts-12hz-1.7b-base
```

**vLLM Version:**
0.25.0
**vLLM-Omni Commit:**
9dd1474b9c042c3061c2f256aa66ad84f24b941e

## Test Result

- Runner Unit Tests: 23 passed
- E2E Tests: 3 passed

### Performance Comparison (This PR vs. Main)
| Concurrency | RTF (PR vs Main) | TTFP ms (PR vs Main) | Throughput (PR vs Main) |
| :--- | :--- | :--- | :--- |
| **1** | 0.246 vs 0.248 (-0.81%) | 177 vs 187 (-5.35%) | 4.092 vs 4.073 (+0.47%) |
| **4** | 0.417 vs 0.416 (+0.24%) | 265 vs 276 (-3.99%) | 9.526 vs 9.572 (-0.48%) |
| **8** | 0.644 vs 0.638 (+0.94%) | 451 vs 467 (-3.43%) | 12.288 vs 12.389 (-0.82%) |
| **16** | 1.104 vs 1.126 (-1.95%) | 1021 vs 1018 (+0.29%) | 14.147 vs 14.095 (+0.37%) |
| **32** | 2.084 vs 2.053 (+1.51%) | 3303 vs 3101 (+6.51%) | 15.363 vs 15.497 (-0.86%) |
| **64** | 3.586 vs 3.614 (-0.77%) | 4590 vs 4903 (-6.38%) | 17.598 vs 17.390 (+1.20%) |

Performance remains neutral across all concurrency levels, with minor fluctuations within expected run-to-run noise margins. Detailed raw JSON results are saved locally and available upon request.

**BEFORE SUBMITTING:** read [CONTRIBUTING.md](https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md) and run the [precheck-pr skill](https://github.com/vllm-project/vllm-omni/blob/main/.claude/skills/precheck-pr/SKILL.md) with the code agent for a self-check against project conventions.
(anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

**Comment by Ronnie-Rui** [comment]
I also found this issue while working on #4745: an explicit speech seed causes talker MTP to fall back to scalar execution.

The graph-stable sampling noise approach makes sense, but there seems to be a correctness gap: `sample_with_exponential_noise()` expects probabilities, while this PR passes raw logits. This computes `argmax(logits / q)`, which is not equivalent to categorical sampling. The logit-domain equivalent should be `argmax(logits - log(q))`, or the logits should be converted to probabilities first.

The new `sampling_noise` path also appears to lack automated coverage. Sampling correctness, seeded batched CUDA Graph, and scalar-versus-batched throughput tests would be useful before merging.

**Comment by sphinxkkkbc** [comment]
**Update**:

1. Added runner unit tests and E2E tests.
2. Added A/B benchmark results against main.

Detailed test plans and benchmark data have been updated in the PR description. Special thanks to @Ronnie-Rui for the 
corrections and suggestions!

Ready for review.
